packages/api-derive/src/assets/assets.spec.ts

```diff
@@ -0,0 +1,238 @@
+            } as unknown as EventRecord] as unknown as Vec<EventRecord>, { createdAtHash: api.createType('Header', { number: new BN(2) }).hash })
+          ]).pipe(concatMap((val) => of(val).pipe(delay(100))))
+        }
+      },
+      rpc: {
+        chain: {
+          subscribeNewHeads: () => from<ObservableInput<Header>>([
+            api.createType('Header', { number: new BN(1) })
+          ])
+        }
+      }
+    } as unknown as DeriveApi;
+  });
+
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+    assets = [];
+    sub.unsubscribe();
+  });
+
+  it('gets current assets list', () => {
+    // eslint-disable-next-line no-return-assign
+    sub = all(mockedApi).subscribe((value) => assets = value);
+
+    expect(assets).toHaveLength(3);
+    expect(assets[0].id.args[0].toNumber()).toBe(FIRST_ASSET_ID);
+    expect(assets[1].id.args[0].toNumber()).toBe(SECOND_ASSET_ID);
+    expect(assets[2].id.args[0].toNumber()).toBe(THIRD_ASSET_ID);
+
+    expect(assets[0].name.toUtf8()).toBe('TestToken');
+    expect(assets[1].name.toUtf8()).toBe('TestTokenExtra');
+    expect(assets[2].name.toUtf8()).toBe('Kusama😻');
+  });
+
+  it('updates assets list after an asset event', () => {
+    // eslint-disable-next-line no-return-assign
+    sub = all(mockedApi).subscribe((value) => assets = value);
+
+    expect(assets).toHaveLength(3);
+
+    jest.advanceTimersByTime(200);
+
+    expect(assets).toHaveLength(2);
+    expect(assets[0].id.args[0].toNumber()).toBe(FIRST_ASSET_ID);
+    expect(assets[1].id.args[0].toNumber()).toBe(THIRD_ASSET_ID);
+
+    expect(assets[0].name.toUtf8()).toBe('TestToken');
+    expect(assets[1].name.toUtf8()).toBe('Kusama😻');
+  });
+
+  it('does not update assets list for a non asset event', () => {
+    const mockedApiWithNonAssetEvent: DeriveApi = {
+      ...mockedApi,
+      query: {
+        ...mockedApi.query,
+        system: {
+          events: () => from<ObservableInput<Vec<EventRecord>>>([
+            [] as unknown as Vec<EventRecord>,
+            Object.assign([{
+              event: {
+                data: [{ module: { index: 34 } }, {
+                  class: 'Normal',
+                  paysFee: 'Yes',
+                  weight: 397453000
+                }] as unknown as GenericEventData,
+                index: api.createType('EventId', '0x0001'),
+                method: 'IdentitySet',
+                section: 'identity'
+              },
+              phase: { ApplyExtrinsic: 1 },
+              topics: [] as unknown as Vec<Hash>
+            } as unknown as EventRecord] as unknown as Vec<EventRecord>, { createdAtHash: api.createType('Header', { number: new BN(2) }).hash })
+          ]).pipe(concatMap((val) => of(val).pipe(delay(100))))
+        }
+      }
+    } as unknown as DeriveApi;
+
+    // eslint-disable-next-line no-return-assign
+    sub = all(mockedApiWithNonAssetEvent).subscribe((value) => assets = value);
```

Comment:
With this change we won't need the eslint-disable.
```suggestion
    sub = all(mockedApiWithNonAssetEvent).subscribe((value) => { assets = value });
```